langlib/lang.runtime/src/main/java/org/ballerinalang/langlib/runtime/Sleep.java

```diff
@@ -35,13 +35,21 @@
 public class Sleep {
 
     private static final int CORE_THREAD_POOL_SIZE = 1;
+    private static final BigDecimal LONG_MAX = new BigDecimal(Long.MAX_VALUE);
 
     private static final ScheduledExecutorService executor = Executors.newScheduledThreadPool(CORE_THREAD_POOL_SIZE);
 
     public static void sleep(Environment env, BDecimal delaySeconds) {
+        BigDecimal delayDecimal = delaySeconds.decimalValue().multiply(new BigDecimal("1000.0"));
+        if (delayDecimal.compareTo(BigDecimal.ZERO) <= 0) {
+            return;
+        }
         Future balFuture = env.markAsync();
-        long delayMillis = (delaySeconds.decimalValue().multiply(new BigDecimal("1000.0"))).longValue();
-        executor.schedule(() -> balFuture.complete(null), delayMillis, TimeUnit.MILLISECONDS);
+        if (delayDecimal.compareTo(LONG_MAX) > 0) {
+            executor.schedule(() -> balFuture.complete(null), Long.MAX_VALUE, TimeUnit.SECONDS);
```

Comment:
Not a big issue, but for correctness this should be TimeUnit.MILLISECONDS as well.